### ISSUE TITLE
Importing a single file with `nixpkgs_package` gives confusing errors

### ISSUE THREAD

**Issue by guibou** [tweag/rules_nixpkgs#99]
By mistake I tried to use `nixpkgs_package` on a nix package which output is a single file, such as:

```
nixpkgs_package(
    name = "output-derivation-is-a-file",
    attribute_path = "",
    nix_file_content = """let pkgs = import <nixpkgs> { config = {}; overlays = []; }; in pkgs.writeText "foo" "bar" """,
    nix_file_deps = [
        "//:nixpkgs.json",
        "//:nixpkgs.nix",
        "//tests:relative_imports/nixpkgs.nix",
    ],
    repository = "@nixpkgs",
)
```

The error message is confusing:

```
$ bazel build @output-derivation-is-a-file//...
INFO: Writing tracer profile to '/home/guillaume/.cache/bazel/_bazel_guillaume/494c5fef8cb6c9d27d36fadb3784eaa6/command.profile.gz'
INFO: Call stack for the definition of repository 'output-derivation-is-a-file' which is a _nixpkgs_package (rule definition at /home/guillaume/tweag/rules_nixpkgs/nixpkgs/nixpkgs.bzl:169:20):
 - /home/guillaume/tweag/rules_nixpkgs/nixpkgs/nixpkgs.bzl:202:9
 - /home/guillaume/tweag/rules_nixpkgs/WORKSPACE:119:1
ERROR: An error occurred during the fetch of repository 'output-derivation-is-a-file':
   java.io.IOException: Could not create symlink from /home/guillaume/.cache/bazel/_bazel_guillaume/494c5fef8cb6c9d27d36fadb3784eaa6/external/output-derivation-is-a-file to /home/guillaume/.cache/bazel/_bazel_guillaume/494c5fef8cb6c9d27d36fadb3784eaa6/external/output-derivation-is-a-file: /home/guillaume/.cache/bazel/_bazel_guillaume/494c5fef8cb6c9d27d36fadb3784eaa6/external/output-derivation-is-a-file (File exists)
ERROR: java.io.IOException: Could not create symlink from /home/guillaume/.cache/bazel/_bazel_guillaume/494c5fef8cb6c9d27d36fadb3784eaa6/external/output-derivation-is-a-file to /home/guillaume/.cache/bazel/_bazel_guillaume/494c5fef8cb6c9d27d36fadb3784eaa6/external/output-derivation-is-a-file: /home/guillaume/.cache/bazel/_bazel_guillaume/494c5fef8cb6c9d27d36fadb3784eaa6/external/output-derivation-is-a-file (File exists)
INFO: Elapsed time: 1.658s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (0 packages loaded)
```

I'd like to be able to import single file like that, but I don't really know how this may fit the bazel paradigm.

However, I think the error message may be more explicit. What do you think? Should I open a PR which checks if the output of the nix derivation is a directory and fail otherwise?

**Comment by regnat** [comment]
I think we could technically make that possible since `new_local_repository` allows that and `nixpkgs_package` is morally a `new_local_repository` applied to the nix-build output.

In the `new_local_repository` case, calling it with a file behaves as if you called it on a directory containing only that file, so calling

```bazel
new_local_repository(
    name = "piano",
    path = "/home/username/Downloads/piano.jar",
    build_file = "BUILD.piano",
)
```

is equivalent to

```bazel
new_local_repository(
    name = "piano",
    path = "/foobar",
    build_file = "BUILD.piano",
)
```

where `foobar` is a directory containing only the `piano.jar` file.

The tricky bit is that in the nix case, the file doesn't really have a meaningful name (since it's a store path), so just doing that wouldn't really make sense.

Bottom line:

> Should I open a PR which checks if the output of the nix derivation is a directory and fail otherwise?

Yes
